x-pack/plugins/search_playground/public/hooks/use_query_indices.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import { IndexName } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { useKibana } from './use_kibana';
+import { ElasticsearchIndex } from '../types';
+
+export const useQueryIndices = (
+  query: string = ''
+): { indices: IndexName[]; isLoading: boolean } => {
+  const { services } = useKibana();
+
+  const { data, isLoading } = useQuery({
+    queryKey: ['indices', query],
+    queryFn: async () => {
+      const response = await services.http.get<{
+        indices: ElasticsearchIndex[];
+      }>('/internal/enterprise_search/indices', {
```

Comment:
:x: You can't use this endpoint here. You'll need to implement what you need in `search_playground/server` this endpoint is not available in Serverless as we do not run the `enterprise_search` plugin there.

Reply:
I suggest we move this to the serverless ticket aswell, as needs some more thought about how to extract this for serverless.